compiler/rustc_feature/src/accepted.rs

```diff
@@ -287,6 +287,8 @@ declare_features! (
     (accepted, const_fn_unsize, "1.54.0", Some(64992), None),
     /// Allows `impl Trait` with multiple unrelated lifetimes.
     (accepted, member_constraints, "1.54.0", Some(61997), None),
+    /// Allows calling `transmute` in const fn
+    (accepted, const_fn_transmute, "1.54.0", Some(53605), None),
```

Comment:
One blocker here is to expand the docs of `transmute`. In particular they need to explain that transmuting pointers to integers is *not* equivalent to casting -- at least during CTFE.

Reply:
Should this be handled for unions as well?

Reply:
Actually just tried transmuting a pointer to an integer in `const fn` with the feature gate enabled. It throws a compilation error stating that it's undefined behavior. Personally I think that's already sufficient, but I can still add it to the API docs if desired.

Reply:
Yes the same is true for unions, but there it's less obvious where to put any docs. ;)

Reply:
If the error is being emitted when it matters (to put it simply), isn't that sufficient? Why duplicate the effort into the documentation when the compiler already does a decent job at it. Helps remove the thinking from the user to an extent as well.

Reply:
Following your logic we should not document when a function `panic`s because "an error is emitted when it matters".

The point of documentation is that users can know how the function behaves *under all circumstances* without having to try it out for themselves.

Reply:
That's fair. I wasn't too set on my position to begin with, and that's more than enough to convince me. I'll add it to the docs.

Reply:
Other than this, the PR is ready. So we need a small stabilization report that is basically "const UB RFC is merged, this is the logical first unsafe const stabilization after that", but in a few more sentences with code examples. I believe there are stab report guidelines in the rustc guide

Reply:
Yeah, was just holding out to wait for CI to be green (before bors cruelly tells me there's now conflicts!) prior to doing that. I'll rebase, add this note in, and write something up. 👍🏽

Reply:
The FCP takes 10 days. You can rebase and fix CI after those 10 days if you want. No need to keep this PR working the whole time

Reply:
Oh I already fixed that locally. Just one conflict. I was just thinking about how to word the docs and where to place it relative to the rest of the documentation on the `transmute` method.

Reply:
Added the following:

> Transmuting pointers to integers in a `const` context is _not_ equivalent to casting using `as`; the resulting integer may not be relied upon for any purpose.